### DISCUSSION TITLE
Implement docfx

### DISCUSSION THREAD

**Discussion by jscarle** [sshnet/SSH.NET#1333] [Ideas]
@drieseng @WojciechNagorski @Rob-Hague 

I'd like to implement [docfx](https://github.com/dotnet/docfx) as a way to automatically build documentation based on the XML docs found in the code. I think this could work really well for this project. The nice thing is that it can automatically be built in a pipeline and then published as a static website using GitHub Pages.

I recently implemented it in my own [LightResults](https://github.com/jscarle/LightResults) repository. After a bit of fighting, I was able to configure it so that it spits out documentation per targeted framework.

As an example, in my repository, the [`IActionableResult`](https://github.com/jscarle/LightResults/blob/main/src/LightResults/Common/IActionableResult.cs) interface is only available starting with .NET 7.0. If you look at the generated documentation, it appears in the [.NET 7.0](https://jscarle.github.io/LightResults/net7.0/LightResults.Common.html) section and the [.NET 8.0](https://jscarle.github.io/LightResults/net8.0/LightResults.Common.html) section, but not in the [.NET 6.0](https://jscarle.github.io/LightResults/net6.0/LightResults.Common.html) section or the [.NET Standard 2.0](https://jscarle.github.io/LightResults/netstandard2.0/LightResults.Common.html) section.

Once the right [configuration](https://github.com/jscarle/LightResults/blob/main/docfx/docfx.json) is made, it's as easy as including a [GitHub Action](https://github.com/jscarle/LightResults/blob/main/.github/workflows/docs.yml) to republish the site whenever there are new changes (or in the case of SSH.NET, new releases).

Before I started work on this, I wanted to make sure that this was something you'd all be on-board with.

**Comment by Rob-Hague** [comment]
Sounds good to me, although I don't know much about building docs. I know a lot of people depend on the CHM file in the releases, but presumably an online version like this would suit them as well.

**Reply by bajpaikunal** [reply]
I was not able to located the CHM file in the release.
